site.py

```diff
@@ -0,0 +1,84 @@
+import os
+import subprocess
+import urllib
+import json
+import time
+import random
+
+from flask import Flask, render_template, url_for, request, redirect, send_from_directory
+
+
+app = Flask(__name__)
+
+# Confirm that a provided file has an acceptable extension.
+# In this case, '.csv' is the only allowed extension.
+def allowed_file(filename):
+    return ('.' in filename) and (filename.rsplit('.', 1)[1] in ALLOWED_EXTENSIONS)
+
+# Retrieve the files uploaded by the user, and if they are valid,
+#  store them in a new generated subfolder of tmp/.
+@app.route('/', methods=['GET', 'POST'])
+def upload_file():
+    if request.method == 'POST':
+        metadatafile = request.files['metadatafile']
+        datafile = request.files['datafile']
+        
+        if metadatafile and metadatafile.filename.endswith('.csv') \
+           and datafile and datafile.filename.endswith('.csv'):
+            print "Saving files..."
+            app.config['UPLOAD_FOLDER'] = 'tmp/'+str(random.randint(10000,99999))+'/'
```

Comment:
This block can be cleaned up. There is no functional requirement to define it on the `app.config`. Also `tmp/` should be defined as a constant in the module. Rather than generating a random integer, generate a UUID.

```python
# Add import at the top
from uuid import uuid4

folder = os.path.join(TEMP_DIR, str(uuid4()))

# Take the basename of the files to ensure no path is included.
metadatafile.save(os.path.join(folder, os.path.basename(metadatafile.filename)))
datafile.save(os.path.join(folder, os.path.basename(datafile.filename)))
```